packages/cluster-ui/src/statementsPage/statementsPage.tsx

```diff
@@ -330,18 +364,32 @@ export class StatementsPage extends React.Component<
     const totalWorkload = calculateTotalWorkload(data);
     const totalCount = data.length;
     const isEmptySearchResults = statements?.length > 0 && search?.length > 0;
+    const nodes = Object.keys(nodeRegions)
+      .map(n => Number(n))
+      .sort();
+    const regions = unique(
+      nodes.map(node => nodeRegions[node.toString()]),
+    ).sort();
+    populateRegionNodeForStatements(statements, nodeRegions);
 
     // Creates a list of all possible columns
     const columns = makeStatementsColumns(
       statements,
       selectedApp,
       totalWorkload,
+      nodeRegions,
       search,
       this.activateDiagnosticsRef,
       onDiagnosticsReportDownload,
       onStatementClick,
     );
 
+    // If it's multi-region, we want to show the Regions/Nodes column by default
+    // and hide otherwise.
+    if (regions.length > 1) {
+      columns.filter(c => c.name === "regionNodes")[0].showByDefault = true;
+    }
+
```

Comment:
Re: my comment above about adding comments, I think this is a nice example to replicate in similar instances.

Reply:
Got it! :)